tests/unit/backend/corpora/common/entities/test_dataset.py

```diff
@@ -86,19 +85,19 @@ def test__create__ok(self):
                 )
 
                 dataset_id = dataset.id
-                artifact_ids = get_ids(dataset.artifacts)
-                deployment_directory_ids = get_ids(dataset.deployment_directories)
-                contributor_ids = get_ids(dataset.contributors)
+                artifact_ids = [i.id for i in dataset.artifacts]
+                deployment_directory_ids = [i.id for i in dataset.deployment_directories]
+                contributor_ids = [i.id for i in dataset.contributors]
 
                 # Expire all local objects and retrieve them from the DB to make sure the transactions went through.
                 Dataset.db.session.expire_all()
 
                 dataset = Dataset.get(dataset_id)
                 self.assertIsNotNone(dataset)
                 self.assertEqual(dataset_id, dataset.id)
-                self.assertEqual(artifact_ids, get_ids(dataset.artifacts))
-                self.assertEqual(deployment_directory_ids, get_ids(dataset.deployment_directories))
-                self.assertEqual(contributor_ids, get_ids(dataset.contributors))
+                self.assertEqual(artifact_ids, [i.id for i in dataset.artifacts])
```

Comment:
Why is testing id equality here important? You could do:

```
expected_artifacts = dataset.artifacts

queried_dataset = Dataset.get(dataset_id)

self.assertCountEqual(queried_dataset.artifacts, expected_artifacts)
```

Something like that?

Reply:
it checks that the rows were linked by verifying the correct IDs were returned